### PR TITLE
controls: push Resume blocking down to car port

### PR DESCRIPTION
**Pull request by jyoung8607**
**Description**

Move the resume block from #25402 down into the GM car port, so that other car ports can use Resume for first engagement.

**Verification**

In progress.

**Route**

Route: TBD

### CODE REVIEW

**Comment by sshane** [comment]
I remember the reason being that this was a feature we wanted to enforce across the cars + it would be too much code to support in GM's interface, but this seems simpler than I thought.

I didn't see any thing about the set speed state machine behavior in 15622:2018, only for ACC. Any thoughts @adeebshihadeh

**Comment by andiradulescu** [comment]
Sorry for spamming this PR, I didn’t realise I was doing it.

I would really love to see this merged!

@adeebshihadeh what do you think?

**Comment by jyoung8607** [comment]
In earlier conversations with the openpilot team, I believe they were open to discussing it. It's in Draft right now because there's a process replay diff with GM, possibly due to different rising/falling edge behavior. If there's going to be a process replay diff, it needs to be fully explained (which I can't yet) and it needs to be tested on a GM car (which I haven't asked anyone to do yet).

**Comment by adeebshihadeh** [comment]
> I remember the reason being that this was a feature we wanted to enforce consistency across the cars + it would be too much code to support in GM's interface, but this seems simpler than I thought.
> 
> I didn't see any thing about the set speed state machine behavior in ISO 15622:2018, only for ACC. Any preference on this @adeebshihadeh?
> 
> In all our cars here, I don't think I've driven one that let's you press the resume button before setting a speed (besides HKG main button), that's probably good to match, unless there's cars that don't do this.

Don't have a strong preference here, aside from generally matching the stock behavior to prevent confusion.

**Comment by sshane** [comment]
@jyoung8607 are you aware of any stock ACC system that allows you to press the resume/+ button on first engage?

**Comment by jyoung8607** [comment]
> @jyoung8607 are you aware of any stock ACC system that allows you to press the resume/+ button on first engage?

Volkswagen. It's the way I (and many others) use openpilot with stock ACC. The current behavior is a hard [POLA](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) violation. 

Can't speak for other cars. If you'd like this PR to support it on a whitelist basis only, that would be reasonable.

File: selfdrive/car/gm/carstate.py [hunk 1]
```diff
@@ -107,6 +108,7 @@ def update(self, pt_cp, cam_cp, loopback_cp):
                       pt_cp.vl["EBCMFrictionBrakeStatus"]["FrictionBrakeUnavailable"] == 1)
 
     ret.cruiseState.enabled = pt_cp.vl["AcceleratorPedal2"]["CruiseState"] != AccState.OFF
+    self.cruise_resume_allowed = self.cruise_resume_allowed or ret.cruiseState.enabled
```
Comment (sshane) [inline selfdrive/car/gm/carstate.py:111]: You also need to consider the main switch https://github.com/commaai/openpilot/pull/908

**Comment by sshane** [comment]
We've moved the car interfacing code to our [`opendbc`](https://github.com/commaai/opendbc) repository, which is now the new home for car ports and fingerprints. Please re-open your pull request against [`opendbc`](https://github.com/commaai/opendbc) at your convenience by using this command below. This will transform all changes under `selfdrive/car/` to `opendbc_repo/opendbc/car/`. Make sure you have initialized submodules beforehand and have checked out the latest `opendbc` commit.

```bash
PR_NUMBER=33045
curl -L https://github.com/commaai/openpilot/pull/$PR_NUMBER.patch | sed -e 's/selfdrive\/car/opendbc_repo\/opendbc\/car/g' | git apply -v --reject
```

Simply replace the PR number with your own. Once done, add the files, fix any conflicts, and open a new PR. Alternatively, you may start a new PR from scratch if that is easier for you.

**Pull request closed without merging**